features/apim-analytics-feature/org.wso2.analytics.solutions.apim.analytics.feature/src/main/resources/siddhi-files/apim_bot_detection_alert_configuration_0.siddhi

```diff
@@ -0,0 +1,59 @@
+/*
+    Copyright (c) 2018, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+@App:name("apim_bot_detection_alert_configuration_0")
+@App:description("Identifies the bot detection")
+
+@sink(type='log')
+@source(type = 'wso2event', wso2.stream.id = 'org.wso2.botDetectedData.request.stream:1.0.0',
+        @map(type = 'wso2event'))
+define stream botDetectionStream(
+    REQUEST_TIME long,
+    MESSAGE_ID string,
+    HTTP_METHOD string,
+    HEADERS string,
+    MESSAGE_BODY string,
+    CLIENT_IP string);
+
+@sink(type='email', content.type="text/html", @map(type ='text', @payload('<strong>Message ID:</strong>{{MESSAGE_ID}} <br><br> <strong>Http method:</strong>{{HTTP_METHOD}} <br><br><strong>Headers:</strong>{{HEADERS}} <br><br> <strong>Message Body:</strong>{{MESSAGE_BODY}} <br><br> <strong>Client IP:</strong>{{CLIENT_IP}}')),subject='Bot detection alert from WSO2 APIM Analytics',to='{{SUBSCRIBER_ADDRESS}}')
+define stream EmailNotificationStream (
+ REQUEST_TIME long,
+ MESSAGE_ID string,
+ HTTP_METHOD string,
+ HEADERS string,
+ MESSAGE_BODY string,
+ CLIENT_IP string,
+ SUBSCRIBER_ADDRESS string
+);
+
+@store(type = 'rdbms', datasource = 'APIM_ANALYTICS_DB')
+@PrimaryKey('MESSAGE_ID')
+define table AM_BOT_DATA (REQUEST_TIME long, MESSAGE_ID string, HTTP_METHOD string, HEADERS string, MESSAGE_BODY string, CLIENT_IP string);
+
+@store(type = 'rdbms', datasource = 'WSO2AM_DB')
+@PrimaryKey('UUID')
+define table AM_NOTIFICATION_SUBSCRIBER(UUID string, CATEGORY string, NOTIFICATION_METHOD string, SUBSCRIBER_ADDRESS string);
+
+from botDetectionStream select * insert into AM_BOT_DATA;
+
+from botDetectionStream as B join AM_NOTIFICATION_SUBSCRIBER as E select B.REQUEST_TIME, B.MESSAGE_ID, B.HTTP_METHOD, B.HEADERS, B.MESSAGE_BODY, B.CLIENT_IP, E.SUBSCRIBER_ADDRESS
```

Comment:
Format as per siddhi query syntax, How much traffic are we expecting for botDetectionStream? Since it is table join per event, this can reduce throughput

Reply:
Very low. Throughput of this stream is not a big concern.